### PR TITLE
Remove mesos lib from pyspark-notebook

### PR DESCRIPTION
**Pull request by parente**
* [x] Remove lib and mentions in the doc
* ~[ ] Add recipe for installing mesos~ I have no way to test a mesos setup. Happy to take a contribution to the recipes page in the documentation if someone would like to pick up that work.

### CODE REVIEW

**Comment by parente** [comment]
@mathbunnyru @romainx rebased and pending Travis results.

File: pyspark-notebook/Dockerfile [hunk 1]
```diff
@@ -24,23 +24,9 @@ RUN cd /tmp && \
...
-RUN apt-get -y update && \
-    apt-get install --no-install-recommends -y gnupg && \
-    apt-key add /tmp/mesos.key && \
-    echo "deb http://repos.mesosphere.io/ubuntu xenial main" > /etc/apt/sources.list.d/mesosphere.list && \
-    apt-get -y update && \
-    apt-get --no-install-recommends -y install mesos=1.2\* && \
-    apt-get purge --auto-remove -y gnupg && \
-    rm -rf /var/lib/apt/lists/*
-
 # Spark and Mesos config
```
Comment (mathbunnyru) [inline pyspark-notebook/Dockerfile:27]: Probably we should remove "mesos" here too.

**Comment by romainx** [comment]
Hello @parente,

I've started the review however I need some additional time since I'm trying to go through the examples of spark cluster connection. If you agree, we could also add some alternative examples showing how to `sparklyr` in R.

**Comment by parente** [comment]
Thanks @romainx. I think it would it be easier to get this PR merged and then to send your proposed documentation changes as a follow-on PR. Does that makes sense? If you see a problem with that approach, let me know and I'll integrate your doc changes directly here.

**Comment by romainx** [comment]
Sure you're right it will be easier. Let's go for the merge.

**Comment by romainx** [comment]
@parente documentation improvement PR for Spark done!

**Pull request merged**